clusterman/cli/status.py

```diff
@@ -109,7 +114,43 @@ def _write_summary(manager: PoolManager) -> None:
     print(f'\tGPUs allocation: {allocated_gpus} GPUs allocated to tasks, {total_gpus} total')
 
 
-def print_status(manager: PoolManager, args) -> None:
+def _json_agent_details(metadata: ClusterNodeMetadata) -> Mapping[str, Any]:
+    return {
+        'agent_state': str(metadata.agent.state),
+        'aws_state': metadata.instance.state,
+        'instance_id': metadata.instance.instance_id,
+        'ip_address': metadata.instance.ip_address,
+        'market': metadata.instance.market,
+        'tasks': metadata.agent.task_count,
+        'uptime': metadata.instance.uptime.total_seconds(),
+        'resources': {
+            resource: {
+                'allocated': getattr(metadata.agent.allocated_resources, resource),
+                'total': getattr(metadata.agent.total_resources, resource),
+            }
+            for resource in ClustermanResources._fields
+        }
+    }
+
+
+def print_status_json(manager: PoolManager, args: argparse.Namespace) -> None:
+    node_metadatas = manager.get_node_metadatas()
+    result = {
+        'disabled': autoscaling_is_paused(manager.cluster, manager.pool, manager.scheduler, arrow.now()),
+        'target_capacity': manager.target_capacity,
+        'fulfilled_capacity': manager.fulfilled_capacity,
+        'non_orphan_fulfilled_capacity': manager.non_orphan_fulfilled_capacity,
+        'resource_groups': [
+            _json_agent_details(metadata)
+            for metadata in node_metadatas
+            for group in manager.resource_groups.values()
+            if metadata.instance.group_id != group.id
+        ],
+    }
+    print(json.dumps(result))
+
+
+def print_status(manager: PoolManager, args: argparse.Namespace) -> None:
```

Comment:
Wouldn't it be better to reimplent this to just parse the dictionary created by `print_status_json`?  Otherwise I worry the two implementations will drift.

Reply:
Yea, fair.  OK.